reporter/src/main/scala/com/typesafe/tools/mima/lib/analyze/Analyzer.scala

```diff
@@ -130,7 +130,7 @@ private[analyze] class TraitAnalyzer extends Analyzer {
   override def analyzeNewClassMethods(oldclazz: ClassInfo, newclazz: ClassInfo): List[Problem] = {
     val res = collection.mutable.ListBuffer.empty[Problem]
 
-    for (newmeth <- newclazz.concreteMethods if !oldclazz.hasStaticImpl(newmeth)) {
+    for (newmeth <- newclazz.concreteMethods if newmeth.isDeferred && !oldclazz.hasStaticImpl(newmeth)) {
```

Comment:
Yeah, sorry, I missed the context (this is in the TraitAnalyzer...). If I'm understanding your code change correctly, you want the loop to be executed only for traits that are compiled with the old encoding. I think it'd be great if you could extract the condition into a method with an intention revealing name :)